### PR TITLE
[SDK] Feature: Pay analytics

### PR DESCRIPTION
**Pull request by gregfromstl**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
The focus of this PR is to add tracking events for various payment methods and transactions within the application.

### Detailed summary
- Added `trackPayEvent` function for tracking payment events
- Updated event tracking in multiple components for different payment modes and transactions
- Enhanced user behavior tracking for payment flows

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/thirdweb/src/analytics/track.ts [hunk 1]
```diff
@@ -20,10 +20,38 @@ export function trackConnect(args: {
...
+  track(args.client, {
+    source: "pay",
+    action: args.event,
+    clientId: args.client.clientId,
+    chainId: args.chainId,
+    walletAddress: args.walletAddress,
+    tokenAddress: args.fromToken,
+    amountWei: args.fromAmount,
+    dstTokenAddress: args.toToken,
+    dstChainId: args.chainId,
```
Comment (joaquim-verges) [inline packages/thirdweb/src/analytics/track.ts:46]: hmmm you prob want to transform these into an existing column we have, or you will have to add new columns to usagev1 table

File: packages/thirdweb/src/analytics/track.ts [hunk 2]
```diff
@@ -20,10 +20,38 @@ export function trackConnect(args: {
...
+  });
+}
+
 /**
  * @internal
  */
 function track(client: ThirdwebClient, data: object) {
+  console.log("tracking", data);
```
Comment (joaquim-verges) [inline packages/thirdweb/src/analytics/track.ts:54]: cleanup

File: packages/thirdweb/src/analytics/track.ts [hunk 3]
```diff
@@ -20,10 +20,38 @@ export function trackConnect(args: {
...
+  fromToken?: string;
+  fromAmount?: string;
+  toToken?: string;
+  toAmount?: string;
+  chainId?: number;
+  dstChainId?: number;
+}) {
+  track(args.client, {
+    source: "pay",
+    action: args.event,
```
Comment (joaquim-verges) [inline packages/thirdweb/src/analytics/track.ts:39]: few things i'd like to have here:

- payMode (can be in the action) - top-up | commerce | transaction
- walletType

File: packages/thirdweb/src/analytics/track.ts (hunk 1 above)
Comment (gregfromstl) [reply packages/thirdweb/src/analytics/track.ts:46]: We have these columns already don't we?

File: packages/thirdweb/src/analytics/track.ts (hunk 3 above)
Comment (gregfromstl) [reply packages/thirdweb/src/analytics/track.ts:39]: Going to track the pay mode in the initial event (when the modal first renders) unless you think we should have it in all events (would have to pass the prop around)

Comment (joaquim-verges) [reply packages/thirdweb/src/analytics/track.ts:40]: for the pay mode, i think we actually already have it in most places via the PayOptions that we pass around no?

File: packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/DirectPaymentModeScreen.tsx [hunk 4]
```diff
@@ -242,6 +243,12 @@ export function DirectPaymentModeScreen(props: {
           variant="accent"
           fullWidth
           onClick={() => {
+            trackPayEvent({
+              event: "choose_payment_method_click",
+              client,
+              walletAddress: payerAccount.address,
+              walletType: activeWallet?.id,
```
Comment (joaquim-verges) [inline packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/DirectPaymentModeScreen.tsx:250]: here we could add pay mode

File: packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/TransactionModeScreen.tsx [hunk 5]
```diff
@@ -225,6 +226,12 @@ export function TransactionModeScreen(props: {
             ) {
               totalCostWei += transactionCostAndData.gasCostWei;
             }
+            trackPayEvent({
+              event: "choose_payment_method_click",
+              client,
+              walletAddress: payerAccount.address,
+              walletType: activeWallet?.id,
```
Comment (joaquim-verges) [inline packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/TransactionModeScreen.tsx:233]: here too

File: packages/thirdweb/src/react/web/ui/TransactionButton/TransactionModal.tsx [hunk 6]
```diff
@@ -27,6 +31,23 @@ export type ModalProps = {
...
+
+  useQuery({
+    queryKey: ["transaction-modal-event"],
+    queryFn: () => {
+      if (!account || !wallet) return;
+      trackPayEvent({
+        client: props.client,
+        walletAddress: account.address,
+        walletType: wallet.id,
+        event: `initiate_${props.payOptions.mode}`,
```
Comment (joaquim-verges) [inline packages/thirdweb/src/react/web/ui/TransactionButton/TransactionModal.tsx:45]: i believe this modal is only for transaction mode

File: packages/thirdweb/src/react/web/ui/TransactionButton/TransactionModal.tsx [hunk 7]
```diff
@@ -27,6 +31,23 @@
...
+
+  useQuery({
+    queryKey: ["transaction-modal-event"],
+    queryFn: () => {
+      if (!account || !wallet) return;
+      trackPayEvent({
+        client: props.client,
+        walletAddress: account.address,
+        walletType: wallet.id,
+        event: "open_pay_modal",
```
Comment (joaquim-verges) [inline packages/thirdweb/src/react/web/ui/TransactionButton/TransactionModal.tsx:45]: open_pay_transaction_modal

**Review by joaquim-verges — APPROVED** [review]

**Pull request merged**